src/Eccube/Form/Type/Admin/MasterdataDataType.php

```diff
@@ -69,6 +69,10 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             if ($data['id'] && empty($data['name'])) {
```

Comment:
please use "==".

Please confirm this test case.
when you input "0" to name and id, this case is OK.